util/logtools/make_timeline.ts

```diff
@@ -0,0 +1,512 @@
+import fs from 'fs';
+import readline from 'readline';
+
+import NetRegexes from '../../resources/netregexes';
+import { NetMatches } from '../../types/net_matches';
+
+import { LogUtilArgParse, TimelineArgs } from './arg_parser';
+import { EncounterCollector, FightEncInfo, TLFuncs } from './encounter_tools';
+
+// TODO: Add support for auto-commenting repeated abilities
+// that can't be synced but should be visible.
+
+// TODO: Add support for assigning sync windows to specific abilities,
+// with or without phase conditions.
+
+// TODO: Add support for auto-generating loops.
+
+// TODO: Add FFLogs report parsing support.
+
+// TODO: Reinstate support for using raw start/end times
+// rather than force fight indexing.
+
+type TimelineEntry = {
+  time: string;
+  combatant?: string;
+  abilityId?: string;
+  abilityName?: string;
+  targetable?: boolean;
+  lineType: string;
+  duration?: number;
+  window?: { beforeWindow: number; afterWindow: number };
+  zoneSeal?: { seal: string; code: '0839' };
+  lineComment?: string;
+};
+
+type ExtendedArgs = TimelineArgs & {
+  [index: string]: string | string[] | number | boolean | undefined;
+  'output-file'?: string;
+  'start'?: string;
+  'end'?: string;
+  'ignore-id'?: string[];
+  'ignore-ability'?: string[];
+  'ignore-combatant'?: string[];
+  'only-combatant'?: string[];
+  'phase'?: string;
+  'include-targetable'?: string[];
+};
+
+// Some NPCs can be picked up by our entry processor.
+// We list them out explicitly here so we can ignore them at will.
+const ignoredCombatants = [
+  'Eos',
+  'Selene',
+  'Garuda-Egi',
+  'Titan-Egi',
+  'Ifrit-Egi',
+  'Carbuncle',
+  'Ruby Ifrit',
+  'Emerald Garuda',
+  'Topaz Titan',
+  'Emerald Carbuncle',
+  'Topaz Carbuncle',
+  'Ruby Carbuncle',
+  'Moonstone Carbuncle',
+  'Rook Autoturret',
+  'Bishop Autoturret',
+  'Demi-Bahamut',
+  'Demi-Phoenix',
+  'Earthly Star',
+  'Seraph',
+  'Automaton Queen',
+  'Bunshin',
+  'Esteem',
```

Comment:
For pet names, maybe it makes sense to pull them in from the auto-generated file instead of duplicating them here?

Reply:
Definitely does, I totally forgot that existed. The generated listing isn't complete, but there certainly is no reason to duplicate the ones that it does contain.

Reply:
If the generated list isn't complete then the tool to generate it should probably be updated?

Reply:
Probably so. It looks like that file hasn't been refreshed since 6.0, so that's probably part of it. As for adding NPCs, I would have to check through the XIVAPI listings to see where they live.

Reply:
This has mostly just been hand-edited by me over time to account for pets and for trust names.  I will say that Estinien is problematic because it skips the Estinien Resentment cast in DSR.

Reply:
That's unfortunate. I guess maybe we could add support for overriding the ignore list? That, or we could just not ignore Estinien and have to remember to manually ignore him for all future dungeons we run with trusts?

Reply:
Yeah, I wouldn't overthink it here.  It's a minor inconvenience we can sort out later with either of those alternatives, I think.